File: android/src/main/kotlin/me/sithiramunasinghe/flutter/flutter_radio_player/core/EventChannelSink.kt
package me.sithiramunasinghe.flutter.flutter_radio_player.core

import io.flutter.plugin.common.EventChannel


class EventChannelSink private constructor() {
    var playbackEventChannel: EventChannel? = null
    var nowPlayingEventChannel: EventChannel? = null
    var playbackVolumeChannel: EventChannel? = null
    companion object {

        @Volatile
        private var instance: EventChannelSink? = null

        fun getInstance(): EventChannelSink {
            if (instance == null) {
                synchronized(this) {
                    if (instance == null) {
                        instance = EventChannelSink()
                    }
                }
            }
            return instance!!
        }
    }
}

File: ios/Classes/core/EventChannelSink.swift
//
//  EventChannelSink.swift
//  flutter_radio_player
//
//  Created by Sithira Munasinghe on 2024-07-23.
//

import Foundation
import Flutter

class EventChannelSink {
    static let instance = EventChannelSink()
    var playbackEventChannel: FlutterEventChannel?;
    var nowPlayingEventChannel: FlutterEventChannel?;
    var playbackVolumeChannel: FlutterEventChannel?;
    
    private init() {
        
    }
    
    
}
